cmd/kubeadm/app/componentconfigs/kubelet.go

```diff
@@ -178,7 +178,7 @@ func (kc *kubeletConfig) Default(cfg *kubeadmapi.ClusterConfiguration, _ *kubead
 	// Let clients using other authentication methods like ServiceAccount tokens also access the kubelet API
 	if kc.config.Authentication.Webhook.Enabled == nil {
 		kc.config.Authentication.Webhook.Enabled = utilpointer.BoolPtr(kubeletAuthenticationWebhookEnabled)
-	} else if *kc.config.Authentication.Webhook.Enabled != kubeletAuthenticationWebhookEnabled {
+	} else if !*kc.config.Authentication.Webhook.Enabled {
```

Comment:
Should this be if *kc.config.Authentication.Webhook.Enabled ?

Reply:
// kubeletAuthenticationWebhookEnabled set the default value to enable authentication webhook
	kubeletAuthenticationWebhookEnabled = true
        if *kc.config.Authentication.Webhook.Enabled != kubeletAuthenticationWebhookEnabled
==   if *kc.config.Authentication.Webhook.Enabled != true
==   if !*kc.config.Authentication.Webhook.Enabled